tavla/src/Shared/hooks/useHeartbeat.ts

```diff
@@ -0,0 +1,204 @@
+                // Format bytes as UUIDv4
+                if (
+                    bytes &&
+                    typeof bytes[6] !== 'undefined' &&
+                    typeof bytes[8] !== 'undefined' &&
+                    bytes.length >= 16
+                ) {
+                    bytes[6] = (bytes[6] & 0x0f) | 0x40
+                    bytes[8] = (bytes[8] & 0x3f) | 0x80
+                    const hex = Array.from(bytes, (b) =>
+                        b.toString(16).padStart(2, '0'),
+                    )
+                    return (
+                        hex.slice(0, 4).join('') +
+                        '-' +
+                        hex.slice(4, 6).join('') +
+                        '-' +
+                        hex.slice(6, 8).join('') +
+                        '-' +
+                        hex.slice(8, 10).join('') +
+                        '-' +
+                        hex.slice(10, 16).join('')
+                    )
+                }
+            }
+        }
+    } catch {}
+
+    //Fallback for old screens
+    const hex = '0123456789abcdef'
+    let uuid = ''
+    for (let i = 0; i < 36; i++) {
+        if (i === 8 || i === 13 || i === 18 || i === 23) uuid += '-'
+        else if (i === 14) uuid += '4'
+        else if (i === 19) uuid += hex[((Math.random() * 4) | 0) + 8]
+        else uuid += hex[(Math.random() * 16) | 0]
+    }
+    return uuid
+}
+
+interface FetchOptions {
+    method?: string
+    headers?: Record<string, string>
+    body?: string
+}
+
+type SafeResponse = { ok: boolean; status: number; text: string }
+
+async function safeFetch(
+    url: string,
+    options: FetchOptions,
+): Promise<SafeResponse> {
+    if (typeof fetch !== 'undefined') {
+        return fetch(url, options).then(async (r) => ({
+            ok:
+                typeof r.ok === 'boolean'
+                    ? r.ok
+                    : r.status >= 200 && r.status < 300,
+            status: r.status,
+            text: await r.text(),
+        }))
+    }
+
+    return new Promise((resolve, reject) => {
+        try {
+            const xhr = new XMLHttpRequest()
+            xhr.open(options.method || 'GET', url, true)
+
+            const headers = options.headers || {}
+            for (const k in headers) {
+                if (Object.prototype.hasOwnProperty.call(headers, k)) {
+                    xhr.setRequestHeader(k, headers[k]!)
+                }
+            }
+
+            xhr.onreadystatechange = function () {
+                if (xhr.readyState === 4) {
+                    resolve({
+                        ok: xhr.status >= 200 && xhr.status < 300,
+                        status: xhr.status,
+                        text: xhr.responseText || '',
+                    })
+                }
+            }
+            xhr.onerror = () => reject(new Error('Network error'))
+            xhr.send(options.body || null)
+        } catch (e) {
+            reject(e)
+        }
+    })
```

Comment:
For å bedre lesbarheten litt -> hva med å dra dette ut i en `xhrFetch`-funksjon? 🤓 

Og kanskje legge på litt `docs` på alle funkjsonene her siden de er litt kryptiske og special-case? 

Feks:

```ts
/**
 * Safely performs a fetch request using either the native `fetch` API or a fallback `xhrFetch` implementation.
 *
 * @param url - The URL to request.
 * @param options - The fetch options, such as method, headers, and body.
 * @returns A promise that resolves to a `SafeResponse` object containing the response status, success flag, and response text.
 */
async function safeFetch(
```